.github/workflows/docs_workflow.yml

```
@@ -0,0 +1,93 @@
+name: CI/CD Release Workflow
+
+on:
+  push:
+    branches: [main]
+
+  release:
+    types: [published]
+
+jobs:
+  publish-docs:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Cache Conda
+        uses: actions/cache@v4
+        env:
+          # Increase this value to reset cache if deploy/conda-dev-spec.template has not changed in the workflow
+          CACHE_NUMBER: 0
+        with:
+          path: ~/conda_pkgs_dir
+          key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{
+            hashFiles('spec-file.txt,deploy/*') }}
```

Since we don't have a `deploy` directory, I think this can just be:
```suggestion
            hashFiles('spec-file.txt') }}
```

---

Trying this now. However, I'm not totally sure what I'm looking for when testing. So far I have been merging into my fork to see how the workflow runs when I merge another PR, but I'm not sure how to tell when it's supposed to be cached or not already.

---

My understanding of this cache stuff is that it won't rebuild the conda environment from scratch if nothing in `spec-file.txt` has changed.  Instead, it will get files from the cache it built the last time it ran, which should save a lot of time.  So we want to identify the file or files that will tell it when it needs to rebuild the conda environment.  Since we build the conda environment from `spec-file.txt`, that should cover us.

In polaris, things are more complicated.  There are several files that control deployment and we point to them all (and a few that don't but it's easier to include them).  If any of them change, we'll rebuild the conda environment to be on the safe side.